gateway/crates/federated-server/src/server/gateway.rs

```diff
@@ -161,11 +170,29 @@ pub(super) async fn generate(
                 tls,
             };
 
-            RedisRateLimiter::runtime(global_config, rate_limiting_configs)
+            match config_path {
+                Some(path) if config_hot_reload => {
+                    hot_reload::ConfigWatcher::new(path, tx).watch()?;
+                }
+                _ => (),
+            }
+
+            RedisRateLimiter::runtime(global_config, rx)
                 .await
                 .map_err(|e| crate::Error::InternalError(e.to_string()))?
         }
-        _ => InMemoryRateLimiter::runtime(KeyedRateLimitConfig { rate_limiting_configs }),
+        _ => {
+            let (tx, rx) = mpsc::channel(100);
+
+            match config_path {
+                Some(path) if config_hot_reload => {
```

Comment:
:bug: **Bug**
The `config_path` and `config_hot_reload` variables are checked for `Some(path)` and `true`, respectively, but there is no handling for the case where `config_path` is `None` or `config_hot_reload` is `false`. This could lead to unexpected behavior if the configuration is not set correctly.

```suggestion
if config_hot_reload { return Err(crate::Error::InternalError("Hot reload is enabled but config_path is not set".into())); }
```